### DISCUSSION TITLE
Simple table block

### DISCUSSION THREAD

**Discussion by beraza** [toeverything/AFFiNE#8618] [Ideas]
### Description

Please add simple table block (like in Notion).
![CleanShot 2024-09-16 at 15 16 36](https://github.com/user-attachments/assets/38b1c9b5-4604-4117-80a1-89e3f4204b24)
![CleanShot 2024-09-16 at 15 17 09](https://github.com/user-attachments/assets/4de50a47-1688-4d02-a513-f2b8c50c0648)

### Use case

It's useful when you don't need the capabilities of a table database, but just need to store information in a tabular format.

### Anything else?

Video preview: https://youtu.be/m_uo9sUvVGA?si=AT_bzDqar65nzAHe

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

[a comment by an automated account is not shown]

**Comment by Flakky** [comment]
Yes, almost every note app has a table block. Very useful feature to have.

**Comment by Sciroccogti** [comment]
And the current Table View does not support copying multiple cells, which I believe is a necessary feature.

**Comment by jghaanstra** [comment]
Even though it's not even there yet I'd also like to see a merge cells feature ...

**Comment by hwangdev97** [comment]
Agree, but we need to consider the long term. Maybe after the API layer of [blocksuite](https://blocksuite.io/) is completed, this could be a third-party block developed by the community?

**Reply by shishkin** [reply]
I'd rather see Table View and Kanban View as third-party blocks. Simple table is bread and butter of document editing.

**Comment by LJungang** [comment]
I am really looking forward to the tabular block feature!

**Comment by djbobyd** [comment]
Even the comment box here supports simple markup tables. Why is it so difficult for Affine to support it as well
| simple | table |
| ---  |  ----  |

**Comment by hwangdev97** [comment]
<img width="1539" alt="image" src="https://github.com/user-attachments/assets/95709bda-c31f-44fa-a1b9-3b5e71a2165b" />
🤌🤌

**Reply by djbobyd** [reply]
That's more like it. It is a very useful component. Thanks a lot for this addition! One small remark if I may, The documents that I write sometimes require to format the text in a cell or even add a formula. Currently formatting is not supported inside the table block, not even simple things. Not to sound ungrateful, but I hope you will add the ability to format the content of the table cells. It seems to be possible as seen below ->

|`some code` | **bold text** |
|-|-|
|_italic text_ | | 

Thanks a bunch again for this addition!
